posthog/models/event/query_event_list.py

```diff
@@ -52,21 +54,21 @@ def query_events_list(
     request_get_query_dict: Dict,
     order_by: List[str],
     action_id: Optional[str],
-    long_date_from: bool = False,
+    select: Optional[List[str]],
+    where: Optional[List[str]],
+    unbounded_date_from: bool = False,
     limit: int = 100,
-) -> List:
+) -> Union[List, dict]:
```

Comment:
This function is 100 lines long and does a lot of:
1. Querying
2. SQL building
3. Post-processing
4. hogql magic

Could we decompose it somehow? E.g. a class-based structure that's similar to other queries?

Reply:
Absolutely! But in a follow up please. I also don't want to prematurely optimise. I just expanded on what was there... 😅 

Somewhat related, I have [another draft PR](https://github.com/PostHog/posthog/pull/13264) out where I'm trying to add HogQL support to existing property filters, and with the upcoming "backend router" work, all of this will be revisited.

Reply:
100% agreed :)